### ISSUE TITLE
Handle duplicate fragments

### ISSUE THREAD

**Issue by mxstbr** [evenchange4/graphql.macro#90]
I have a query that uses a fragment directly, and another fragment that uses the same fragment in a nested manner:

```graphql
query getViewerCommunities($id: ID!) {
  user(id: $id) {
    ...userInfo
    communities {
      participants {
        ...communityParticipantInfo
      }
    }
  }
}
${userInfoFragment}
${communityParticipantInfoFragment}
```

With a `communityParticipantInfo` fragment that looks like this:

```graphql
fragment communityParticipantInfo on CommunityParticipant {
  idm
  user {
    ...userInfo
  }
}
${userInfoFragment}
```

As you can see, the `userInfoFragment` is repeated twice. Using `graphql-tag` at runtime correctly deduplicates that fragment and everything works perfectly.

Unfortunately, `graphql.macro` does not deduplicate the fragment and my server thusly throws an error saying "Duplicate fragment" and rejects the query entirely.

**Comment by FezVrasta** [comment]
@evenchange4 are you going to work on this? I can give it a shot otherwise

**Comment by evenchange4** [comment]
Verified in my project. It should be fixed with version `v1.4.2` now. Thanks @FezVrasta !

**Comment by danielkcz** [comment]
@evenchange4 I am using 1.4.2 and this is still the issue. Fragments are duplicated and the server rejects it. I've checked the test you added for this and I don't think it's sufficient. Apparently, something else is happening when fragments are nested over several documents. I've verified that with repeated fragments over a single documents the deduplication indeed works.

```gql
query {
  order {
    items {
      ...FItemDetail
      subitems {
         ...FItemDetail
      }
    }
  }
}
```

```gql
fragment FItemDetail on Item {
  price {
     ...FPrice
  }
}
```

```gql
fragment FPrice on PriceConverted {
  value
  currency
}
```
